go/merkletree2/tree.go

```diff
@@ -388,6 +437,25 @@ type MerkleInclusionProof struct {
 	RootMetadataNoHash  RootMetadata `codec:"e"`
 }
 
+// A MerkleExtensionProof proves, given the RootMetadata hashes of two merkle
+// trees and their respective Seqno values, that: - the two merkle trees have
+// the expected Seqno values, - the most recent merkle tree "points back" to the
+// least recent one through a chain of SkipPointers that refer to merkle trees
+// at intermediate Seqnos.
+type MerkleExtensionProof struct {
+	_struct              struct{}       `codec:",toarray"` //nolint
+	RootHashes           []Hash         `codec:"h"`
+	PreviousRootsNoSkips []RootMetadata `codec:"k"`
+}
+
+// An MerkleInclusionExtensionProof combines a MerkleInclusionProof and a
+// MerkleExtensionProof. The redundant fields are deleted so that sending a
+// combined proof is more efficient than sending both of them individually.
+type MerkleInclusionExtensionProof struct {
```

Comment:
Do you ever serialize this and send it over the wire? This feels brittle. I don't trust when the fancy features of go (anonymous inclusion) interface with msgpack.

Reply:
Curious to know how this is encoded.

Concat'ed or a pair of two arrays?

Reply:
It should be a pair of two arrays. It would have been concatenated with the anonymous fields of the previous commit (which is why I had done it that way).

Reply:
Nope, it just concatenates them together. I tested it

Reply:
Oh, wait, I tested the anonymous fields, did this change?